### ISSUE TITLE
[override] Does not replace tag in 1.29.5

### ISSUE THREAD

**Issue by tulku** [influxdata/telegraf#14885]
### Relevant telegraf.conf

```toml
[agent]
  interval = "1s"
  flush_interval = "1s"

[[inputs.file]]
    files = ["/test.in"]

    [inputs.file.tags]
        bucket_name = "original"

[[processors.override]]
    namepass = ["should_change_bucket"]
    [processors.override.tags]
        bucket_name = "changed"

    [processors.override.tagdrop]
      monitor = [ "*" ]

[[outputs.file]]
  files = ["stdout"]
```

### Logs from Telegraf

```text
2024-02-22T19:08:00Z I! Loading config: /etc/telegraf/telegraf.conf.d/tag_override.conf
2024-02-22T19:08:00Z I! Starting Telegraf 1.29.5 brought to you by InfluxData the makers of InfluxDB
2024-02-22T19:08:00Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-02-22T19:08:00Z I! Loaded inputs: file
2024-02-22T19:08:00Z I! Loaded aggregators: 
2024-02-22T19:08:00Z I! Loaded processors: override
2024-02-22T19:08:00Z I! Loaded secretstores: 
2024-02-22T19:08:00Z I! Loaded outputs: file
2024-02-22T19:08:00Z I! Tags enabled: host=3b029d3f686c
2024-02-22T19:08:00Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"3b029d3f686c", Flush Interval:1s
should_keep_bucket,bucket_name=original,host=3b029d3f686c,id=42 humidity=35i 1708627414000000000
should_change_bucket,bucket_name=original,host=3b029d3f686c,id=43 temperature=25 1708627414000000000
```

### System info

Telegraf 1.29.5 in -alpine docker image, host ubuntu 22.04

### Docker

FROM telegraf:1.29.5-alpine
COPY ./telegraf_config/override_bug/test.in /
COPY ./telegraf_config/override_bug /etc/telegraf/telegraf.conf.d

CMD telegraf --config-directory /etc/telegraf/telegraf.conf.d/

### Steps to reproduce

1. Run the provided configuration, on telegraf 1.29.5 using the `test.in` specified below.
3. Observe the metrics written to stdout.

### Expected behavior

If we do change the telegraf version to 1.29.4 in the Dockerfile, and use the same input file, the generated metrics are:
```
should_keep_bucket,bucket_name=original,host=f46e5b469754,id=42 humidity=35i 1708627414000000000
should_change_bucket,bucket_name=changed,host=f46e5b469754,id=43 temperature=25 1708627414000000000
```
Notice how the `bucket_name` tag is different in the two metrics, the first one has the unchanged value `original`, while the second one was changed to `changed` as it is the intent of the configuration.

(full logs)
```
2024-02-22T19:06:16Z I! Loading config: /etc/telegraf/telegraf.conf.d/tag_override.conf
2024-02-22T19:06:16Z I! Starting Telegraf 1.29.4 brought to you by InfluxData the makers of InfluxDB
2024-02-22T19:06:16Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-02-22T19:06:16Z I! Loaded inputs: file
2024-02-22T19:06:16Z I! Loaded aggregators: 
2024-02-22T19:06:16Z I! Loaded processors: override
2024-02-22T19:06:16Z I! Loaded secretstores: 
2024-02-22T19:06:16Z I! Loaded outputs: file
2024-02-22T19:06:16Z I! Tags enabled: host=f46e5b469754
2024-02-22T19:06:16Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"f46e5b469754", Flush Interval:1s
should_keep_bucket,bucket_name=original,host=f46e5b469754,id=42 humidity=35i 1708627414000000000
should_change_bucket,bucket_name=changed,host=f46e5b469754,id=43 temperature=25 1708627414000000000
...
```

### Actual behavior

The `bucket_name` tag value is not changed, and both output metrics have the same tag value:
```
should_keep_bucket,bucket_name=original,host=3b029d3f686c,id=42 humidity=35i 1708627414000000000
should_change_bucket,bucket_name=original,host=3b029d3f686c,id=43 temperature=25 1708627414000000000
```

### Additional info

test.in file:
```
should_keep_bucket,id=42 humidity=35i 1708627413906535248
should_change_bucket,id=43 temperature=25 1708627413906535248
```

**Comment by srebhan** [comment]
@tulku this should be already fixed by #14880 can you please check if the binary in the PR fixes the issue!?

**Comment by tulku** [comment]
Yes! The binary in the linked pr fixes the issue:
```
2024-02-22T20:07:57Z I! Loading config: telegraf_config/override_bug/tag_override.conf
2024-02-22T20:07:57Z I! Starting Telegraf 1.30.0-a268d206 brought to you by InfluxData the makers of InfluxDB
2024-02-22T20:07:57Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 61 outputs, 6 secret-stores
2024-02-22T20:07:57Z I! Loaded inputs: file
2024-02-22T20:07:57Z I! Loaded aggregators: 
2024-02-22T20:07:57Z I! Loaded processors: override
2024-02-22T20:07:57Z I! Loaded secretstores: 
2024-02-22T20:07:57Z I! Loaded outputs: file
2024-02-22T20:07:57Z I! Tags enabled: host=wyvern
2024-02-22T20:07:57Z I! [agent] Config: Interval:1s, Quiet:false, Hostname:"wyvern", Flush Interval:1s
should_keep_bucket,bucket_name=original,host=wyvern,id=42 humidity=35i 1708627414000000000
should_change_bucket,bucket_name=changed,host=wyvern,id=43 temperature=25 1708627414000000000
```
The tag `bucket_name` changes as expected.

Thanks!
